ivy/functional/ivy/gradients.py

```diff
@@ -180,7 +193,11 @@ def stop_gradient(x, preserve_type=True):
 
 
 @to_native_arrays_and_back
-def execute_with_gradients(func, xs, retain_grads=False):
+def execute_with_gradients(
+    func: Callable,
+    xs: Union[ivy.Array, ivy.NativeArray, ivy.Container],
```

Comment:
we should not include `ivy.Container` in the type hints of any nestable functions.